### PR TITLE
a_teammate/stackwalker

### PR DESCRIPTION
**Pull request by a-teammate**
this pull request enables that if inexor crashes, it displays a list of lastly executed commands.

on linux and mac it puts that info to the command line output (and the log-file ofc), on windows it additonally pops up another window on crash,

### CODE REVIEW

File: inexor/crashreporter/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,18 @@
+
+if(NOT MINGW)
+declare_module(crashreporter .)
+add_lib(crashreporter)
+endif()
+
+function(config_crashreporter targ)
+  if(MINGW)
+    return()
+  endif()
```
Comment (koraa) [inline inexor/crashreporter/CMakeLists.txt]: why no mingw?

File: inexor/crashreporter/StackWalker_windows.cpp [hunk 2]
```diff
@@ -0,0 +1,1025 @@
+///////////////////////////////////////////////////////////////////////////
```
Comment (koraa) [inline inexor/crashreporter/StackWalker_windows.cpp]: Why do we need 1k lines for windows?

File: inexor/engine/main.cpp [hunk 3]
```diff
@@ -1198,10 +1199,18 @@ ICOMMAND(cef_reload, "", (),
 ICOMMAND(cef_focus, "b", (bool *b),
     if (cef_app.get()) cef_app->setFocus(*b); );
 
+#include <csignal>
```
Comment (koraa) [inline inexor/engine/main.cpp]: Is this for debugging? Please remove.

Comment (a-teammate) [inline inexor/engine/main.cpp]: its for testing the crashreporter and will be removed as last step before a merge

File: inexor/crashreporter/StackWalker_windows.cpp (hunk 2 above)
Comment (a-teammate) [inline inexor/crashreporter/StackWalker_windows.cpp]: yeah .. :/ i didnt want to write the platform stuff myself so this part is actualyl a lib

**Comment by a-teammate** [comment]
going to merge in 1h

**Comment by koraa** [comment]
Seems OK to me. Needs a bit of refactoring, but we can probably merge.

**Comment by koraa** [comment]
Please apply my comments though.

**Comment by a-teammate** [comment]
merged

**Comment by Fohlen** [comment]
This still hangs up on linux when using ```/killprogramm``` instead properly exiting.
Please investigate/fix.

**Pull request closed without merging**